File: ios/HybridRiveImage.swift
import NitroModules
import RiveRuntime

class HybridRiveImage: HybridRiveImageSpec {
  let renderImage: RiveRenderImage
  let rawData: Data

  init(renderImage: RiveRenderImage, rawData: Data) {
    self.renderImage = renderImage
    self.rawData = rawData
    super.init()
  }

  var byteSize: Double {
    Double(rawData.count)
  }
}

File: android/src/legacy/java/com/margelo/nitro/rive/HybridRiveImage.kt
package com.margelo.nitro.rive

import androidx.annotation.Keep
import app.rive.runtime.kotlin.core.RiveRenderImage
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridRiveImage(
  val renderImage: RiveRenderImage,
  private val dataSize: Int
) : HybridRiveImageSpec() {
  override val byteSize: Double
    get() = dataSize.toDouble()
}
